### PR TITLE
BUILD: a couple of cmake msvc compile fixes

### PR DESCRIPTION
**Pull request by cjj19970505**
Throw error when using multi-config generators.
Add USE_MSVC_STATIC_CRT option for statically linking with CRT.
Set curl's CURL_STATIC_CRT aligned with USE_MSVC_STATIC_CRT.
Fix cmake configs of 7z and OpenAL in msvc build.
Remove unecessary hard coded compiling and linking flags.
Use debug option in windeployqt when configuring for a debug build.
Search for windeployqt tool in configure time.
Add vcpkg's Qt support.

This PR enables CMake build support for
Release build with both static CRT and dynamic CRT
Debug build with dynamic CRT

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -29,17 +29,64 @@ set(CMAKE_MODULE_PATH "${CMAKE_CURRENT_LIST_DIR}/buildfiles/cmake")
...
+    option(USE_MSVC_STATIC_CRT "Use static MSVC C runtime" OFF)
+
+    # TODO(cjj19970505@live.cno)
+    # DiscordRPC binary in 3rdparty is compiled /MT
+    # So theoretically we should enable DiscordRPC in Release and static CRT build
+    # since we might encounter some rumtime issues when more than one CRT version are presented.
+    # https://docs.microsoft.com/en-us/cpp/c-runtime-library/crt-library-features?view=msvc-160#what-problems-exist-if-an-application-uses-more-than-one-crt-version
+    # Add other DiscordRPC binaries(compiled with /MTd, /MD, /MDd) or compile it from source may address this issue.
+    if(NOT(CMAKE_BUILD_TYPE MATCHES "Release" AND USE_MSVC_STATIC_CRT))
+        set(USE_DISCORD_RPC OFF CACHE BOOL "Discrd RPC is only availiable in Relase and static CRT build." FORCE)
```
Comment (Megamouse) [inline CMakeLists.txt:58]:
```suggestion
        set(USE_DISCORD_RPC OFF CACHE BOOL "Discord RPC is only available in Release and static CRT build." FORCE)
```

File: rpcs3/CMakeLists.txt [hunk 2]
```diff
@@ -149,9 +149,48 @@ elseif(UNIX)
...
+    # this is a single-config generator workaround
+    # multi-config generator will fail.
+
+    # TODO(cjj19970505@live.cn)
+    # offical Qt binaries are built with -MD(d) only as stated in offical wiki
+    # https://wiki.qt.io/Technical_FAQ#Why_does_a_statically_built_Qt_use_the_dynamic_Visual_Studio_runtime_libraries_.3F_Do_I_need_to_deploy_those_with_my_application_.3F
+    # If we build our libs with /MT(d), we might encounter some issues.
+    # https://docs.microsoft.com/en-us/cpp/c-runtime-library/crt-library-features?view=msvc-160#what-problems-exist-if-an-application-uses-more-than-one-crt-version
+
+    # Qt installed from Qt installer hs following hierarchy:
```
Comment (Megamouse) [inline rpcs3/CMakeLists.txt:162]:
```suggestion
    # Qt installed from Qt installer has following hierarchy:
```

File: rpcs3/CMakeLists.txt [hunk 3]
```diff
@@ -149,9 +149,48 @@ elseif(UNIX)
...
+    # bin/ for release and debug dlls and windeployqt tools
+    # lib/cmake/Qt5/ for Qt5_Dir
+    # Qt installed from vcpkg has following hierarchy:
+    # bin/ for release dlls
+    # debug/bin/ for debug dlls
+    # tools/qt5/bin/ for tools including windeployqt
+    # tools/qt5/debug/bin/ for tools with debug build including windeployqt
+    # share/cmake/Qt5/ for Qt5_Dir
+
+    # If Qt5 is installed from official Qt instaler
```
Comment (Megamouse) [inline rpcs3/CMakeLists.txt:172]:
```suggestion
    # If Qt5 is installed from official Qt installer
```

**Review by hcorion — APPROVED** [review]
Not sure about functionality, since I don't use MSVC. but this looks good from a code perspective

**Pull request merged**